From 8eac87f0a10f8e241fe8972527069ae6bbd36429 Mon Sep 17 00:00:00 2001
From: Stefan Assmann <sassmann@redhat.com>
Date: Tue, 2 Sep 2014 13:10:54 -0400
Subject: [ethernet] i40evf: remove VLAN filters on close

Message-id: <1409663591-32531-12-git-send-email-sassmann@redhat.com>
Patchwork-id: 90489
O-Subject: [RHEL7.1 PATCH 011/148] i40evf: remove VLAN filters on close
Bugzilla: 1091127
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Mitch Williams <mitch.a.williams@intel.com>

We remove all the MAC filters, so remove the VLAN filters, too.

Change-ID: I4f7559acdf005dc3f359bf6460ce32d183c8878b
Signed-off-by: Mitch Williams <mitch.a.williams@intel.com>
Signed-off-by: Jesse Brandeburg <jesse.brandeburg@intel.com>
Tested-by: Sibai Li <sibai.li@intel.com>
Signed-off-by: Aaron Brown <aaron.f.brown@intel.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Stefan Assmann <sassmann@redhat.com>
(cherry picked from commit ed1f5b58ea01757c5878ecff762e884767da9bab)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/i40evf/i40evf_main.c b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
index 1629b73..75a2c6f 100644
--- a/drivers/net/ethernet/intel/i40evf/i40evf_main.c
+++ b/drivers/net/ethernet/intel/i40evf/i40evf_main.c
@@ -972,9 +972,14 @@ void i40evf_down(struct i40evf_adapter *adapter)
  list_for_each_entry(f, &adapter->mac_filter_list, list) {
   f->remove = true;
  }
+ /* remove all VLAN filters */
+ list_for_each_entry(f, &adapter->vlan_filter_list, list) {
+  f->remove = true;
+ }
  if (!(adapter->flags & I40EVF_FLAG_PF_COMMS_FAILED) &&
      adapter->state != __I40EVF_RESETTING) {
   adapter->aq_required |= I40EVF_FLAG_AQ_DEL_MAC_FILTER;
+  adapter->aq_required |= I40EVF_FLAG_AQ_DEL_VLAN_FILTER;
   /* disable receives */
   adapter->aq_required |= I40EVF_FLAG_AQ_DISABLE_QUEUES;
   mod_timer_pending(&adapter->watchdog_timer, jiffies + 1);
-- 
1.7.1